ui/widgets/controlgroup.js

```diff
@@ -0,0 +1,242 @@
+			"selectmenu": "select",
+			"spinner": ".ui-spinner-input",
+			"controlgroupLabel": ".ui-controlgroup-label"
+		},
+		direction: "horizontal",
+		excludeInvisible: true
+	},
+
+	_create: function() {
+		this._enhance();
+	},
+
+	// To support the enhanced option in jQuery Mobile, we isolate DOM manipulation
+	_enhance: function() {
+		this.element.attr( "role", "toolbar" );
+		this.refresh();
+	},
+
+	_destroy: function() {
+		this._callChildMethod( "destroy" );
+		this.childWidgets.removeData( "ui-controlgroup-data" );
+		this.element.removeAttr( "role" );
+	},
+
+	_initWidgets: function() {
+		var that = this,
+			childWidgets = [];
+
+		// First we iterate over each of the items options
+		$.each( this.options.items, function( widget, selector ) {
+			var widgets, labels,
+				options = {};
+
+			// We assume everything is in the middle to start because we can't determine
+			// first / last elements until all enhancments are done.
+			if ( that[ "_" + widget + "_options" ] ) {
+				options = that[ "_" + widget + "_options" ]( "middle" );
+			}
+
+			// Make sure the widget actually exists and has a selector set
+			if ( $.fn[ widget ] && selector ) {
+
+				// Find instances of this widget inside controlgroup and set options
+				widgets = that.element.find( selector )[ widget ]( options );
+
+				widgets.each( function() {
+					var element = $( this ),
+
+					// Set data on the widget element pointing to the this.element of the widget
+					// and telling us what type of widget this is
+					widgetElement = element[ widget ]( "widget" ).data(
+						"ui-controlgroup-data",
+						element.data( "ui-" + widget.charAt(0).toUpperCase() + widget.slice(1) )
+					);
+
+					childWidgets.push( widgetElement[ 0 ] );
+				} );
+			} else if ( selector && widget === "controlgroupLabel" ) {
+				labels = that.element.find( selector );
+				labels.each( function() {
+					$( this ).contents().wrapAll( "<span class='ui-controlgroup-label-contents'></span>" );
+				} );
+				that._addClass( labels, null, "ui-widget ui-widget-content ui-state-default" );
+				Array.prototype.push.apply( childWidgets, labels.get() );
+			}
+		} );
+
+		this.childWidgets = $( $.unique( childWidgets ) );
+		this._addClass( this.childWidgets, "ui-controlgroup-item" );
+	},
+
+	_callChildMethod: function( method ) {
+		this.childWidgets.each( function() {
+			var element = $( this ),
+				data = element.data( "ui-controlgroup-data" );
+			if( data && data[ method ] ) {
+				data[ method ]();
+			}
+		} );
+	},
+
+	_updateCornerClass: function( element, position ) {
+		var direction = this.options.direction === "vertical"
+			remove = "ui-corner-top ui-corner-bottom ui-corner-left ui-corner-right",
+			add = this._buildSimpleOptions( position, direction, "label" ).classes.label;
+
+		this._removeClass( element, null, remove );
+		this._addClass( element, null, add );
+	},
+
+	_buildSimpleOptions: function( position, direction, key ) {
```

Comment:
I suggest we remove all instances of the `direction` argument/parameter. In the two methods where its actually needed (here and `_selectmenu_options`), it can be replaced by direct access to `this.options.direction`. That simplifies all the other `*_options` methods and their callers.